File: Tweeter.WebApi/Tweeter.Application/Models/TweetComment.cs
using System;

namespace Tweeter.Application.Models
{
	public class TweetCommentDto
	{
		public int Id { get; set; }
		public string Text { get; set; }
		public string ImgUrl { get; set; }
		public int TweetId { get; set; }
		public int LikeCount { get; set; }
		public bool IsLikedByCurrentUser { get; set; }
		public DateTime CreatedAt { get; set; }
		public UserDto CreatedBy { get; set; }
	}

	public class TweetCommentForCreation
	{
		public string Text { get; set; }
		public int TweetId { get; set; }
		public File Img { get; set; }
	}

	public class TweetCommentSearchParam : PageRequest
	{
		public int? CurrentUserId { get; set; }
		public int? TweetId { get; set; }
		public int? IdLessThan { get; set; }
	}
}

File: tweeter-app/src/app/models/TweetComment.ts
import { ApiPageRequest } from './Api';
import { User } from './User';

export class TweetComment {
    public id: number;
    public text: string;
    public imgUrl: string;
    public tweetId: number;
    public likeCount: number;
    public isLikedByCurrentUser: boolean;
    public createdAt: Date;
    public createdBy: User;
    public img: File;
}

export class TweetCommentSearchParam extends ApiPageRequest {
    public tweetId: number;
    public idLessThan: number;
}
